openmls/src/binary_tree/test_binary_tree.rs

```diff
@@ -147,63 +156,54 @@ fn test_diff_merging() {
     assert_eq!(tree, original_tree);
 }
 
-#[test]
-fn test_leaf_addition_and_removal_errors() {
-    let tree = MlsBinaryTree::new((0..3).collect()).expect("error creating tree");
-    let mut diff = tree.empty_diff();
-
-    diff.remove_leaf().expect("error removing leaf");
-
-    // Should fail removing the last remaining leaf.
-    assert_eq!(
-        diff.remove_leaf()
-            .expect_err("no error trying to remove the last leaf in the diff"),
-        MlsBinaryTreeDiffError::TreeTooSmall
-    );
-
-    // Let's test what happens when the tree is getting too large.
-    let mut nodes: Vec<u32> = Vec::new();
-
-    // We allow uninitialized vectors because we don't want to allocate so much memory
-    #[allow(clippy::uninit_vec)]
-    unsafe {
-        nodes.set_len(u32::MAX as usize);
-
-        let tree = MlsBinaryTree::new(nodes).expect("error creating tree");
-        let mut diff = tree.empty_diff();
-
-        assert_eq!(
-            diff.add_leaf(666, 667)
-                .expect_err("no error adding beyond u32 max"),
-            MlsBinaryTreeDiffError::TreeTooLarge
-        )
-    }
-}
-
```

Comment:
Is there a particular reason why this test is removed?

Reply:
The unsafe code no longer works, because we now do something with the nodes. But I think I can re-add the removal check.